app/actions/handlers.py

```diff
@@ -58,23 +58,43 @@ def get_inaturalist_observations(integration: Integration, config: PullEventsCon
                        "uri", "photos", "user", "location", "place_ids", "taxon", "photos.large_url", "photos.url", "taxon.id", "taxon.rank", "taxon.name",
                        "taxon.preferred_common_name", "taxon.wikipedia_url", "taxon.conservation_status", "user.id", "user.name", "user.login",
                        "annotations.controlled_attribute_id", "annotations.controlled_value_id"])
-
-    inat_count_req = get_observations_v2(page = 1, per_page = 0, updated_since = since, project_id = config.projects, quality_grade = config.quality_grade,
-                                      taxon_id=target_taxa, nelat = float(nelat), nelng = float(nelng), swlat = float(swlat), swlng = float(swlng),
-                                      order_by = "updated_at", order="asc")
-
+    get_observations_params = { "page": 1,
+                                "per_page": 0,
+                                "updated_since": since,
+                                "project_id" : config.projects,
+                                "quality_grade" : config.quality_grade,
+                                "taxon_id" : target_taxa, 
+                                "order_by" : "updated_at", 
+                                "order" : "asc"}
+    if nelat and nelng and swlat and swlng:
+        get_observations_params["nelat"] = nelat
+        get_observations_params["nelng"] = nelng
+        get_observations_params["swlat"] = swlat
+        get_observations_params["swlng"] = swlng
+    inat_count_req = get_observations_v2(get_observations_params)
```

Comment:
you need to use the unpacking operator `**` here
```suggestion
    inat_count_req = get_observations_v2(**get_observations_params)
```